packages/xstate-store/src/types.ts

```diff
@@ -109,10 +111,10 @@ export interface Store<
    */
   trigger: {
     [E in TEvent as E['type'] & string]: IsEmptyObject<
-      Omit<E, 'type'>
+      DistributiveOmit<E, 'type'>
     > extends true
-      ? () => Omit<E, 'type'>
-      : (eventPayload: Omit<E, 'type'>) => void;
+      ? () => DistributiveOmit<E, 'type'>
```

Comment:
it's unrelated to this PR but this looks like a mistake, it should be:
```suggestion
      ? () => void
```